translate/filters/checks.py

```diff
@@ -2476,6 +2476,36 @@ def nplurals(self, unit):
 
         return True
 
+    @critical
+    def untranslated_plurals(self, unit):
+        """Check if any of the plural targets is translated while other isn't.
+
+        This check fails if the language's nplurals is different than the
+        unit's plural targets count. This is so because otherwise no error
+        would be reported if for example a unit has more plural targets than
+        its language's nplurals, but only has translated exactly the same
+        number of plural targets as the language's nplurals.
+        """
+        if not unit.hasplural():
+            return True
+
+        nplurals = self.config.lang.nplurals
+
+        if not nplurals > 0:
+            return True
+
+        if len(unit.target.strings) != nplurals:
+            # Report as error if unit has a number of targets different from
+            # its language nplurals.
+            return False
+
+        translated_count = 0
+        for target in unit.target.strings:
+            if len(target) > 0:
+                translated_count += 1
+
+        return translated_count == 0 or translated_count == nplurals
```

Comment:
I'd rather raise SeriousFilterFailure and allow things to be commented such as Nth plural form is missing.

It might be simpler for readers if this is broken up into the two different things that are being checked.  First you're checking that its untranslated in all forms.  Then you're checking that the translated forms are all translated.  Just break that apart and humans will be able to follow it.